lib/matplotlib/ticker.py

```diff
@@ -1258,19 +1283,17 @@ def format_eng(self, num):
 
         mant = sign * dnum / (10 ** pow10)
 
+        # TODO: shouldn't we raise a warning if self.places < 0?
```

Comment:
More or less done, but there is still the issue of 2 remaining failing tests (see my comment below).